controllers/topics-controllers.js

```diff
@@ -1,22 +1,33 @@
 const {
-    selectTopics,
+    selectArticles,
     selectArticleById,
     updateArticleById,
+    selectTopics,
     selectUsers,
-    selectCommentsByArticle,
 } = require('../models/topics-models.js');
 
-exports.getTopics = (req, res, next) => {
-    selectTopics().then((topics) => {
-        res.status(200).send({ topics });
-    });
-};
 
-exports.getUsers = (req, res, next) => {
-    selectUsers().then((users) => {
-        res.status(200).send({ users });
+exports.getArticles = (req, res, next) => {
+    selectArticles().then((articleArray) => {
+        const idArray = [];
+        articleArray.forEach((article) => {
+            idArray.push(article.article_id);
+        })
+        function getCount(array, value) {
+            return array.filter((v) => (v === value)).length;
+        }
+        const articles = [articleArray[0]]
+        articleArray.forEach((article) => {
+            if (article.article_id !== articles[0].article_id) {
+                articles.unshift(article)
+            }
+        })
+        articles.forEach((article) => {
+            article.comment_count = getCount(idArray, article.article_id);
```

Comment:
This kind of work to interact with the data to extract a comment count out should definitely be in the model BUT it also feels like a lot more leg work than I'd expect - did you consider doing a COUNT() in your PSQL query with a join with the comments table? Let me know if this doesn't make sense to you but it's cut out so much leg work if you created comment_count in your PSQL query

Reply:
I didn't know COUNT() was a thing! Thanks, I've refactored the article models and controllers to use this now. Wish I'd tried this before, it was much quicker and looks tidier I think.